src/utils.ts

```diff
@@ -145,29 +146,77 @@ export const getCioClient = (apiKey?: string, cioJsClientOptions?: ConstructorCl
 
 export const getActiveSectionsWithData = (
   activeSections: UserDefinedSection[],
-  sectionResults: SectionsData
+  sectionResults: SectionsData,
+  sectionsRefs: React.MutableRefObject<React.RefObject<HTMLLIElement>[]>
 ) => {
   const activeSectionsWithData: Section[] = [];
-  activeSections?.forEach((config) => {
-    const { identifier } = config;
-    let data;
+  activeSections?.forEach((sectionConfig, index) => {
+    const { identifier } = sectionConfig;
+    let sectionData;
 
-    if (isCustomSection(config)) {
+    if (isCustomSection(sectionConfig)) {
       // Copy id from data to the top level
-      data = config.data.map((item) => ({
+      sectionData = sectionConfig.data.map((item) => ({
         ...item,
         id: item?.id || item?.data?.id,
       }));
     } else {
-      data = sectionResults[identifier];
+      sectionData = sectionResults[identifier];
     }
 
-    if (Array.isArray(data)) {
-      activeSectionsWithData.push({ ...config, data });
+    if (Array.isArray(sectionData)) {
+      const section = {
+        ...sectionConfig,
+        data: sectionData,
+      };
+
+      // If user provided a ref in `sectionConfig`, use it. Otherwise, use the ref from our library generated refs array
```

Comment:
Four questions:
1. how will the lib consumer know that they can pass these refs? 
2. how will lib consumer know how to pass this ref? 
3. what happens if they don't pass this ref and are using the lib's component interface? 
4. what happens if they don't pass this ref and are using the lib's hook interface?

Reply:
1. It's a prop on type `SectionConfiguration` https://github.com/Constructor-io/constructorio-ui-autocomplete/pull/107/files/55c168e54f3bc1568827cba615592e193f2fc3c1#diff-c54113cf61ec99691748a3890bfbeb00e10efb3f0a76f03a0fd9ec49072e410aR83
2. same as 1 but we can add more documentation explaining that
3. and 4. There's no change between hooks and component interface as long as they use getSectionProps

We need to add docs for getSectionProps

Reply:
This is a great callout. This would have made things more clear if added from the beginning. Added here
https://github.com/Constructor-io/constructorio-ui-autocomplete/pull/107/commits/06ced4f6e422f6213c02d08570acb4abba0f084c#diff-8fa4b52909f895e8cda060d2035234e0a42ca2c7d3f8f8de1b35a056537bf199R40 @deomsj